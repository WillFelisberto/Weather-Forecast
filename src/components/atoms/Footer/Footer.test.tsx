import { render, screen } from '@/tests/test-utils';

import { Footer } from '.';

describe('Footer Component', () => {
  it('renders the correct text', () => {
    const { container } = render(<Footer />);
    expect(screen.getByText(/Made with/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('contains a link to GitHub', () => {
    render(<Footer />);
    const link = screen.getByRole('link', { name: /Willian Souza/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', 'https://github.com/willFelisberto/');
  });
});
