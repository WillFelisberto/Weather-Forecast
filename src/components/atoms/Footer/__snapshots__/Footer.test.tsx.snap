// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders the correct text 1`] = `
<div>
  <footer
    class="w-full py-4 text-center mt-auto"
  >
    <p
      class="text-white text-sm"
    >
      Made with 
      <span
        class="text-red-500"
      >
        &lt;3
      </span>
       by
      <a
        class="text-blue-400 hover:underline ml-1"
        href="https://github.com/willFelisberto/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Willian Souza
      </a>
    </p>
  </footer>
</div>
`;
